by-language/java-quarkus-panache/src/main/java/io/crate/CrateDbDialect.java

```diff
@@ -0,0 +1,115 @@
+package io.crate;
+
+import org.hibernate.boot.model.TypeContributions;
+import org.hibernate.dialect.DatabaseVersion;
+import org.hibernate.dialect.PostgreSQLDialect;
+import org.hibernate.dialect.sequence.NoSequenceSupport;
+import org.hibernate.dialect.sequence.SequenceSupport;
+import org.hibernate.service.ServiceRegistry;
+import org.hibernate.type.descriptor.jdbc.spi.JdbcTypeRegistry;
+import org.hibernate.type.descriptor.sql.internal.DdlTypeImpl;
+import org.hibernate.type.descriptor.sql.spi.DdlTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hibernate.type.SqlTypes.*;
+
+/**
+ * extends postgres sql dialect with crate specialities and also cover limitations
+ *
+ * @author mackerl
+ */
+public class CrateDbDialect extends PostgreSQLDialect {
+
+    private static final Logger log = LoggerFactory.getLogger(CrateDbDialect.class);
+
+    public CrateDbDialect() {
+        super(DatabaseVersion.make(14));
+        log.info("creating crate dialect");
+    }
+
+    @Override
+    public SequenceSupport getSequenceSupport() {
+        return NoSequenceSupport.INSTANCE;
+    }
+
+    @Override
+    protected void registerColumnTypes(TypeContributions typeContributions, ServiceRegistry serviceRegistry) {
+        super.registerColumnTypes(typeContributions, serviceRegistry);
+        final DdlTypeRegistry ddlTypeRegistry = typeContributions.getTypeConfiguration().getDdlTypeRegistry();
+
+        // Prefer object if possible
+        ddlTypeRegistry.addDescriptor(new DdlTypeImpl(JSON, "OBJECT(DYNAMIC)", this));
+    }
+
+    @Override
+    protected String columnType(int sqlTypeCode) {
+        switch (sqlTypeCode) {
```

Comment:
What's the difference of overriding this method vs overriding registration in the constructor like: https://github.com/manuelbernhardt/hibernate-core/blob/master/hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQLDialect.java#L57

Reply:
don't really know ... derived from https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQLDialect.java